python/Scripts/createdefinition.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python
+'''
+Create definitions based on the nodegraphs in the specified document. Definitions are 
+saved to new documents with either the definition and functional graph in a single file or
+in separated into a file for definitions and a file for functional graphs.
+
+It is assumed that all nodegraphs have the same category, nodegroup, version and namespace
+as only one set of these input options can be provided. It is also assumed that each nodegraph
+has a different output signature. If the same signature if out then no definition will be created
+for that nodegraph.
```

Comment:
Perhaps we should omit this Python example, since it's relatively limited in its target user base?  If we have teams asking for this functionality in the future, we can propose it again, but for now it seems to be putting the cart before the horse!

Reply:
Hi @jstone-lucasfilm,

With new nodes coming on line, and folks are hand editing them or using some custom nodedef generation that requires a DCC, it seems that utility is of use. How about I move this to a new PR, to get some feedback
and conversation as to what criteria and logic is of interest.

( Aside: I would actually prefer the previously brought up idea of have a Python "examples" or "utilities" library added.  The command line scripts are okay, but have overlap for some functionality which would be useful to pull out. )

What do you think ?

Reply:
@kwokcb I'm definitely interested in providing good examples of publishing workflows, which can at first be used directly by users in open-source tools like MaterialXGraphEditor, and eventually used as a template when proprietary tools implement their own publishing strategies.

So far, though, it's not clear that Python scripts will be a part of these workflows, and my initial sense is that we want to focus on the API level functionality (as the remainder of this pull request does), and then determine the best integration into MaterialXGraphEditor (as your counterpart pull request does).

In order to facilitate each of these steps, my recommendation would be to make each pull request as minimal as possible, so that we can hammer out the details of the required functionality that each next step will build upon.  In hammering out those details, I think it's likely that the direction of the steps that follow will be significantly altered, so there's a huge advantage in keeping them as simple as possible.

Reply:
Sounds good. I've remove the sample script.